### DISCUSSION TITLE
JSON feed with post tags support?

### DISCUSSION THREAD

**Discussion by gouku** [facebook/docusaurus#6322] [Q&A] [answered]
I'm glad now it has JSON feed support so I can fetch posts programmatically: https://github.com/facebook/docusaurus/pull/6126

But it seems the JSON feed output does not contain tags. Any easy way how to add/hook to support this?

**Comment by slorber** [comment]
One way would to be to write your own json file, by extending the blog plugin: https://github.com/facebook/docusaurus/issues/4138#issuecomment-771042127

IE you'd be responsible for converting a blog post into the json file, like we do here:

```js
    async postBuild({outDir}: Props) {
      if (!options.feedOptions.type) {
        return;
      }

      // TODO: we shouldn't need to re-read the posts here!
      // postBuild should receive loadedContent
      const blogPosts = await generateBlogPosts(contentPaths, context, options);
      if (!blogPosts.length) {
        return;
      }
      await createBlogFeedFiles({
        blogPosts,
        options,
        outDir,
        siteConfig,
      });
    },
```

It may not be convenient in the current state (cf TODO): we should provide `content.blogPosts` as param to `postBuild`

Does it make sense?

**Reply by gouku** [reply]
Thanks! This should work. But it would be great if Docusaurus has tags support natively since tags are part of the built-in feature.

**Reply by Josh-Cena** [reply]
Indeed it's something we can have. JSON feed actually have this in their spec: https://www.jsonfeed.org/version/1/

In the [feed](https://github.com/jpmonette/feed) library we use, this seems to be called `category`

**Reply by gouku** [reply]
> In the [feed](https://github.com/jpmonette/feed) library we use, this seems to be called `category`

Thansk for replying. But the `category` fields seems not currently available in the latest build?

**Reply by Josh-Cena** [reply]
Yes, that's just a note to myself about how we will implement it :)

**Comment by Josh-Cena** [comment, marked as answer]
https://github.com/facebook/docusaurus/pull/6388 this has been implemented!
